image/v1/tests/ImageStreamImportMode.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "Image"
+crdName: images.config.openshift.io
+tests:
+  onCreate:
+   - name: Should be able to create a minimal Image config
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec: {}
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec: {}
+        status:
+          internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+   - name: Should be able to configure imageStreamImportMode to Legacy
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec:
+          imageStreamImportMode: Legacy
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec:
+          imageStreamImportMode: Legacy
+        status:
+          internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+          imageStreamImportMode: Legacy
+   - name: Should be able to configure imageStreamImportMode to PreserveOriginal
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec:
+          imageStreamImportMode: PreserveOriginal
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec:
+          imageStreamImportMode: PreserveOriginal
+        status:
+          internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
```

Comment:
Is this a defaulted value?

Reply:
this value comes from the image-registry service by concatenating the <name>.<namespace>.svc:port and is the default for a cluster.

Reply:
When writing the tests here, you must only add fields like this that are defaulted by the API itself. I don't think this one is API defaulted so I think this will break your tests presently

Reply:
removed that field.